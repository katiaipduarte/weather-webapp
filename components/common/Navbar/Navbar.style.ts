import styled from 'styled-components'

export const NavbarContainer = styled.nav`
	z-index: 999;

	.menu {
		display: flex;
		flex-wrap: wrap;
		justify-content: flex-start;
		align-items: center;
		list-style-type: none;

		ul {
			li a {
				display: block;
				padding: 1.5rem 0.5rem;
				text-decoration: none;
			}
		}

		.logo {
			font-size: 0.75rem;
			font-weight: 600;
			mix-blend-mode: difference;
		}

		.item {
			padding: 1rem;
			text-align: center;
			font-size: 1.25rem;
			color: white;
			mix-blend-mode: difference;
		}
	}
`
