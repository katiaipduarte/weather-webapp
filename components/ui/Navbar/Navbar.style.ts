import styled from 'styled-components'

export const NavbarContainer = styled.header`
	.menu {
		display: flex;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;

		.menu {
			position: relative;

			li {
				a {
					display: block;
					padding: 0 0.5rem;
					text-decoration: none;
				}
			}
		}

		.toggle {
			padding: 1.5rem 1rem;
			order: 1;
			a {
				svg {
					color: white;
					mix-blend-mode: difference;
					height: 1.5rem;
					font-size: 2rem;
				}
			}
		}

		.logo {
			color: white;
			mix-blend-mode: difference;
			padding: 1.5rem 1rem;
		}

		.item,
		.mobile-item {
			color: white;
			padding: 1rem;
			order: 3;
			width: 100%;
			text-align: center;
			display: none;
			font-size: 1.25rem;
		}
	}

	.active {
		.item {
			display: none;
		}

		.mobile-menu {
			position: absolute;
			top: 4rem;
			left: 0;
			right: 0;
			z-index: 2;
			background-color: rgba(0, 0, 0, 0.95);
			-webkit-backdrop-filter: blur(10px);
			backdrop-filter: blur(10px);

			.mobile-item {
				display: block;
			}
		}
	}

	/* Desktop menu */
	@media all and (min-width: 960px) {
		.menu {
			justify-content: center;
			align-items: flex-start;
			flex-wrap: nowrap;
			background: none;
			width: 65%;
			.logo {
				flex: 1;
				order: 0;
			}
			.toggle {
				display: none;
			}
			.item {
				display: block;
				width: auto;
				font-size: 1rem;
				border: none;
			}
		}
	}
`
