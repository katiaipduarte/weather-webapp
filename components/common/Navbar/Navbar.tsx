import Link from 'next/link'

import { NavbarContainer } from './Navbar.style'

const Navbar = (): JSX.Element => {
	return (
		<NavbarContainer>
			<ul className='menu' data-testid='menu'>
				<li className='logo'>
					<Link href='/'>weather.app‚Å¢</Link>
				</li>
				<li className='item'>
					<Link href='/favourites'>Favourites</Link>
				</li>
				<li className='item'>
					<Link href='/search-history'>History</Link>
				</li>
				{/* <li className='toggle' data-testid='menu-btn'>
					<a
						role='button'
						tabIndex={0}
						aria-label='Click to open mobile menu'
						onClick={() => setSeeMobileMenu(!seeMobileMenu)}
						data-testid='mobile-menu-btn'
					>
						<FontAwesomeIcon icon={faBars} />
					</a>
				</li> */}
			</ul>
		</NavbarContainer>
	)
}

export default Navbar
