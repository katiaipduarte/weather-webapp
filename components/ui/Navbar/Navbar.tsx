import SearchBar from '@components/common/SearchBar/SearchBar'
import { faBars } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Link from 'next/link'
import { useState } from 'react'

import { NavbarContainer } from './Navbar.style'

const Navbar = (): JSX.Element => {
	const [seeMobileMenu, setSeeMobileMenu] = useState<boolean>(false)

	return (
		<NavbarContainer>
			<nav>
				<ul
					data-testid='menu'
					className={seeMobileMenu ? 'menu active' : 'menu'}
				>
					<li className='logo'>
						<Link href='/'>weather.app‚Å¢</Link>
					</li>
					<li className='toggle' data-testid='menu-btn'>
						<a
							role='button'
							tabIndex={0}
							aria-label='Click to open mobile menu'
							onClick={() => setSeeMobileMenu(!seeMobileMenu)}
							data-testid='mobile-menu-btn'
						>
							<FontAwesomeIcon icon={faBars} />
						</a>
					</li>
					<li className='item'>
						<Link href='/favourites'>Favourites</Link>
					</li>
					<li className='item'>
						<Link href='/search-history'>History</Link>
					</li>

					<ul className='mobile-menu'>
						<li className='mobile-item'>
							<Link href='/favourites'>Favourites</Link>
						</li>
						<li className='mobile-item'>
							<Link href='/search-history'>History</Link>
						</li>
						<li className='mobile-item'>
							<SearchBar />
						</li>
					</ul>
				</ul>
			</nav>
		</NavbarContainer>
	)
}

export default Navbar
