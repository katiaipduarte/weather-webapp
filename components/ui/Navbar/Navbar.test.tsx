import { fireEvent, render, screen } from '@testing-library/react'
import Navbar from './Navbar'

describe('<Navbar /> component Unit Test', () => {
	it('should render 1 <Navbar /> component', () => {
		const { container } = render(<Navbar />)
		expect(container).toMatchSnapshot()
	})

	it('should add active to className in ul, when clicking the faBars icon', () => {
		render(<Navbar />)

		const menuBtn = screen.getByTestId('menu-btn')
		const menu = screen.getByTestId('menu')

		expect(menuBtn).toBeTruthy()
		expect(menu.className).toBe('menu')

		fireEvent.click(screen.getByTestId('mobile-menu-btn'))
		expect(menu.className).toBe('menu active')
	})
})
