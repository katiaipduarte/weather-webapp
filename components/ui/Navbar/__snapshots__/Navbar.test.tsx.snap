// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> component Unit Test should render 1 <Navbar /> component 1`] = `
<div>
  <header
    class="sc-dkzDqf cAbPCq"
  >
    <nav>
      <ul
        class="menu"
        data-testid="menu"
      >
        <li
          class="logo"
        >
          <a
            href="/"
          >
            weather.app‚Å¢
          </a>
        </li>
        <li
          class="toggle"
          data-testid="menu-btn"
        >
          <a
            aria-label="Click to open mobile menu"
            data-testid="mobile-menu-btn"
            role="button"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-bars "
              data-icon="bars"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            href="/favourites"
          >
            Favourites
          </a>
        </li>
        <li
          class="item"
        >
          <a
            href="/search-history"
          >
            History
          </a>
        </li>
        <ul
          class="mobile-menu"
        >
          <li
            class="mobile-item"
          >
            <a
              href="/favourites"
            >
              Favourites
            </a>
          </li>
          <li
            class="mobile-item"
          >
            <a
              href="/search-history"
            >
              History
            </a>
          </li>
          <li
            class="mobile-item"
          >
            <form
              aria-label="Search location form"
              class="sc-gsnTZi hMnVqC"
              role="searchbox"
              tabindex="-1"
            >
              <button
                aria-label="Click to search"
                id="search-term-btn"
                tabindex="-1"
                type="submit"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass "
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <input
                aria-label="Another location"
                maxlength="50"
                placeholder="Another location"
                role="search"
                type="text"
                value=""
              />
              <button
                aria-label="Click to clear search"
                id="clear-search-term-btn"
                style="visibility: hidden;"
                tabindex="0"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-xmark "
                  data-icon="xmark"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </form>
          </li>
        </ul>
      </ul>
    </nav>
  </header>
</div>
`;
